### PR TITLE
fix(Flag): allow hex to be passed in as bg prop

### PR DESCRIPTION
**Pull request by msafari**
problem:
if we passed in hex values directly to Flag, the UI was kinda broken.
screen shot 2018-04-24 at 9 13 33 am

fixed:
screen shot 2018-04-24 at 9 33 55 pm

### CODE REVIEW

**Comment by tornglintaffychen** [comment]
Thanks for updating.

File: src/Flag.js [hunk 1]
```diff
@@ -7,28 +7,36 @@ import Box from './Box'
 
 const capitalize = str => str.charAt(0).toUpperCase() + str.slice(1)
 
-const darkBorderColor = props => {
+const shadowColor = props => {
   const darkColor = theme(`colors.dark${capitalize(props.color)}`)(props)
+
+  if (!darkColor) {
```
Comment (msafari) [inline src/Flag.js]: props to @jxnblk for your suggestion. This is much better than adding a darkBgColor prop 🙇‍♀️

File: src/Flag.js [hunk 2]
```diff
@@ -7,28 +7,36 @@ import Box from './Box'
...
+      `
+    }
+  }
+
   return {
-    borderTopColor: darkColor,
-    borderRightColor: darkColor
+    backgroundImage: `linear-gradient(45deg, transparent 50%, ${darkColor} 50%)`
   }
 }
```
Comment (BeniCheni) [inline src/Flag.js]: Could these two `return` statements consolidate into one `return` statement?
```
return !darkColor
  ? { ... }
  : { ... }
```
Or the ternary goes inside the returned `backgroundImage` object key, as it seems the ternary conditions return the same `backgroundImage` object key?
```
return {
  backgroundImage: !darkColor ? `...` : `...`
   }
```

Which one would you like better, if you decide to consider consolidating?

Comment (msafari) [reply src/Flag.js]: I updated it to
```
  return {
    backgroundImage: !darkColor
      ? `
        linear-gradient(45deg, transparent 50%, rgba(0, 0, 0, 0.5) 50%),
        linear-gradient(45deg, transparent 50%, ${props.color} 50%)
      `
      : `linear-gradient(45deg, transparent 50%, ${darkColor} 50%)`
  }
```

File: src/Flag.js [hunk 3]
```diff
@@ -7,28 +7,32 @@ import Box from './Box'
...
+  ${shadowColor};
   position: absolute;
   bottom: 0;
 `
 
 const FlagRight = styled(Box)`
   flex: none;
-  background-color: ${props => theme(`colors.${props.color}`)(props)};
+  background-color: ${props =>
+    theme(`colors.${props.color}`, props.color)(props)};
```
Comment (jxnblk) [inline src/Flag.js]: :+1: good use of the fallback argument here

**Comment by jxnblk** [comment]
LGTM - feel like if this used an SVG instead of CSS hacks, this component would probably be a little easier to work on

File: src/Flag.js (hunk 3 above)
Comment (msafari) [reply src/Flag.js]: lol thanks for your suggestion. I didn't know about it :D

**Pull request merged**